File: Backend/Python/Data Structure/sort/binary_search.py
def binarySearch(target, sortedLyst):
    """二分查找"""
    left = 0
    right = len(sortedLyst) - 1
    while left <= right:
      mid_index = (left + right) // 2
      if target == sortedLyst[mid_index]:
        return mid_index
      elif target < sortedLyst[mid_index]:
        right = mid_index - 1
      else:
        left = mid_index + 1
    return -1

File: Software Engineering/iOS/Swift/Data Structure/Search/BinarySearch.swift
import Foundation

final class BinarySearch {

    // t: O(log(n)), s: O(1)
    static func solution<T: Comparable>(input: [T], target: T) -> Int? {

        var left = 0
        var right = input.count

        while left < right {
            let midIndex = (right + left) / 2
            let midNum = input[midIndex]

            if midNum == target {
                return midIndex
            } else if midNum < target {
                left = midIndex + 1
            } else {
                right = midIndex
            }
        }

        return nil
    }

}
